### PR TITLE
Changes to include source tag to xrpl tx

### PR DESCRIPTION
**Pull request by KarishmaBothara**
This PR should include a sourceTag for XRPL withdraw (bridging TRN->XRPL) tx.

### CODE REVIEW

File: pallet/xrpl-bridge/Cargo.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ serde = { version = '1.0.144', features = ['derive'], optional = true }
 sp-runtime = { git = 'https://github.com/paritytech/substrate', branch = "polkadot-v0.9.30", default-features = false }
 sp-std = { git = 'https://github.com/paritytech/substrate', branch = "polkadot-v0.9.30", default-features = false }
 sp-core = { git = 'https://github.com/paritytech/substrate', branch = "polkadot-v0.9.30", default-features = false }
-xrpl-codec = { git = "https://github.com/futureversecom/xrpl-tx-codec", rev = "0.0.9" }
+xrpl-codec = { git = "https://github.com/futureversecom/xrpl-tx-codec", branch = "support/sourceTag", default-features = false }
```
Comment (surangap) [inline pallet/xrpl-bridge/Cargo.toml:29]: can merge the xrpl-codec PR, bump and update to use here

File: pallet/xrpl-bridge/src/lib.rs [hunk 2]
```diff
@@ -105,6 +105,8 @@ pub mod pallet {
 		/// Threshold to emit event TicketSequenceThresholdReached
 		type TicketSequenceThreshold: Get<Percent>;
 
+		type SourceTag: Get<u32>;
```
Comment (surangap) [inline pallet/xrpl-bridge/src/lib.rs:108]: Hope this is going to be a constant.
```suggestion
                #[pallet::constant]
		type SourceTag: Get<u32>;
```

**Review by surangap — COMMENTED** [review]
lgtm! might require a local test to verify all good.

**Comment by KarishmaBothara** [comment]
> lgtm! might require a local test to verify all good.

so, I did a local test by setting door account, next threshold sequence and everything..

and did a withdraw

Screenshot 2023-11-22 at 12 09 22 PM

```
 const blob = '0x12000022800000002302515fdb2400000000202901d80ffc6140000000000000076840000000000f4240730081143216fd40be8f9b0016253e5244085375d887a53e831416a5e24f755c0b2e6ecac217b0b052921fb5d000';
    console.log(JSON.stringify(xrpl.decode(blob)));
    console.log('*********')

```

```
{"TransactionType":"Payment","Flags":2147483648,"SourceTag":38887387,"Sequence":0,"TicketSequence":30937084,"Amount":"7","Fee":"1000000","SigningPubKey":"","Account":"rnZiKvrWFGi2JfHtLS8kxcqCqVhch6W5k5","Destination":"rshkbs2J5RkwDeD79MhJ5yU7KYhbU3YyEm"}
*********
```

Also when I decode the blob, I get the source tag as expected.

To test it, we would need to deploy on porcini to get the signature to reach the sign quorum.

**Review by surangap — APPROVED** [review]
lgtm👍

File: pallet/xrpl-bridge/src/lib.rs (hunk 2 above)
Comment (KarishmaBothara) [reply pallet/xrpl-bridge/src/lib.rs]: Made it flexible... can be set via extrinsic

File: pallet/xrpl-bridge/src/lib.rs [hunk 3]
```diff
@@ -225,6 +221,11 @@ pub mod pallet {
 	/// Highest settled XRPL ledger index
 	pub type HighestSettledLedgerIndex<T: Config> = StorageValue<_, u32, ValueQuery>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn xrp_source_tag)]
```
Comment (surangap) [inline pallet/xrpl-bridge/src/lib.rs:241]: try not to define getters for storage items. we try to move away from using it. can use `SourceTag::<T>::get()` for this. gives more clarity when reading the code.

**Review by surangap — APPROVED** [review]
lgtm

**Pull request merged**